packages/headless/src/api/knowledge/createAnswerAPISlice.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {ConfigurationSection} from '../../state/state-sections';
+import {createAbortController} from '../../utils/abort-controller-polyfill';
+import {baseKnowledgeAPISlice} from './baseKnowledgeAPISlice';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+const abortController = createAbortController();
+let retryCount = 0;
+
+const data = {
```

Comment:
`data` (and `type Data`) is a poor name.

How about `GeneratedAnswerStream` ?

Reply:
data IS a poor name. :+1:  thanks for pointing it out